### DISCUSSION TITLE
Checkpointing and resuming after a system crash

### DISCUSSION THREAD

**Discussion by hariseldon99** [jcmgray/quimb#250] [Q&A]
Hi,

What is the best way to checkpoint-to-disk a TEBD run of quimb in case my cluster crashes (due to power outages) and I need to resume later? Is it optimal to write a dictionary containing the timestamp, Hamiltonian in symbolic form, and the full psi state at the timestamp in the spin basis (although the curse of dimensionality might need to be addressed)?

**Comment by jcmgray** [comment]
Hi @hariseldon99, you could try simply saving the whole `tebd` object to disk:
```python
qu.save_to_disk(tebd, "tebd.dmp")
tebd = qu.load_from_disk("tebd.dmp")
```
which is just a simple alias for `joblib.dump`. Whilst its not the most universal or parseable format I think, it should serve checkpointing purposes fine. Let me know if you have any trouble!
